### DISCUSSION TITLE
too many updates and the process is not convenient enough to be a charm process

### DISCUSSION THREAD

**Discussion by jgarces-pd** [dbeaver/dbeaver#19763] [Ideas]
### Description

Every time that I open DBeaver Community, every time that I get a new update.

I'm not against to updates, I actually like it when devs cares a lot to deliver constant updates, but it must be a better way to deliver it...

Each time that I need to apply an update, I need to download the full binary again. Why don't just make a background update, like every other product (e.g. opera browser, or any other linux binary), then use the newest version the next time that you open the app?

### DBeaver Version

CE Version 23.0.2.202304091457

### Operating System

macOS Ventura 13.3.1 (22E261)

**Comment by mattfrey-dna** [comment]
@jgarces-pd I agree with you. I think DBeaver is an awesome product, but I have the same feelings regarding the constant updates. Perhaps there could be a way to set _always download and update_ the software or a choice to _opt out_ of this. Usually, the alert disrupts my work flow and it's kind of a pain. All in all, great product tho.

**Comment by mpbits** [comment]
I am glad I am not the only one who is annoyed by this process. There must be a better way. Maybe backgroung update process like Chrome or monthly.

**Comment by tslouf-umich** [comment]
Product is good. Active development is great. More notifications is annoying. If we have to be notified, it might be nice to specify which updates we want to see. Major versions 2x or minor 23.x rather than every micro release of 23.x.yzabcde.

**Comment by MiiChielHD** [comment]
I have the same 'frustration'. I'm an active developer, and I don't want to spend time closing DBeaver soo many times, waiting for it too update... and the next week again.

The product is great, I'm gratefull for the active development, yet less updates in a relative short amount of time would be more usuable. 

Except if the previous update broke something ;)

**Comment by khanshahan** [comment]
Agree with all of you. Have been using DBeaver for ages now. Love this product. But the amount of updates is just too many. Background service is not ideal for digital nomads (the growing trend). Many work with the cellular service or with internet that may not be very good. Hence, downloading in background reduces performance of work.  While EU is still very good with internet service, UK is horrible and super backward.

Perhaps possibility of bundled monthly updates with an option of allow the updates to be downloaded whenever possible?
A long term plan perhaps can be to build the architecture in a way that updates can be installed in background without disrupting the work and take affect when dbbeaver is next restarted.

**Reply by MiiChielHD** [reply]
Agreed, I even prefer a 3 monthly update instead... The start-up time of DBeaver is pretty slow, even on a i9-9900K... So even the extra updates just don't help.

If you look at how Discord does his update policy, directly when booting the application and the updating time is around 5 seconds.

**Comment by devdiogenes** [comment]
I got same opinion, in fact, I found this discussion searching about how to disable check for updates.

**Comment by MiiChielHD** [comment]
I think the best solution is that the binary is downloaded in the backgrounded, when succesfull unpacked, then when the application starts, or a pre application that the user starts without knowing, always checks if there is an updated version downloaded. If so the updated version is copied to the working app folder and then executed. If no updated version is downloaded, the already installed version in the working app folder is executed.

Problem solved.

**Comment by regoake** [comment]
I couldn't agree more. There's no way a mature product needs this many updates to maintain functionality and security so it's a lot of hassle for no apparent gain. As many have mentioned, an automated process where the latest version is launched without having to annoy users and interrupt workflow would be very much appreciated.

**Comment by joekrukosky** [comment]
Me too. Even checking the no more reminders for this version doesn't help much. If nothing else, please change the reminder to a passive popup so I don't have to click every time.

**Comment by SCdF** [comment]
I love that DBeaver is actively worked on. 

Having said that, apart from security issues or critical bug fixes, a lower release cadence of even once a month would improve the usability significantly. At the moment, especially because it's an app I open once every couple of weeks to debug something, nearly every time I open it I get asked to update.

**Comment by jamnum1** [comment]
+2

**Comment by ged-yuko** [comment]
DBeaver has a long story of fighting for autoupdates, here are some eposides:
https://github.com/dbeaver/dbeaver/issues/2444
https://github.com/dbeaver/dbeaver/issues/4791
https://github.com/dbeaver/dbeaver/issues/5317
https://github.com/dbeaver/dbeaver/issues/6522
https://github.com/dbeaver/dbeaver/issues/6558
https://github.com/dbeaver/dbeaver/issues/12172
https://github.com/dbeaver/dbeaver/issues/18112

**Comment by macr1408** [comment]
You can just go to preferences -> User Interface -> disable "Automatic updates check"

**Reply by MiiChielHD** [reply]
Although that temporarily solves the issues, it prevents retreiving important updates against vulnerabilities. 

I would advice the developersteam to look at Discord, they either have automatic update when you start the application, or an icon in the upper right when an update is ready. Making it optional to automatically update when starting the application would be important in situations a system administrator needs to fix something quickly without being delayed by the auto-update.

Also the update should be just one click, that starts from the main window of DBeaver, no full dialog, just a click, it updates itself and you are good to go.

**Comment by JackLingwood** [comment]
There are too many DBeaver updates. If this continues I will start looking at other tools.

**Comment by Sam-VO** [comment]
Even making the update flow better would be a big step forward. https://github.com/dbeaver/dbeaver/issues/40014
- Ask to install updates when you close the app
- Move the update notification to a in app banner which can be acted on when it's convenient
- Auto start the app after an update installs

Forcing the user to update right when the app opens is the worst time to ask because they opened the app to do something which wasn't check for updates. Updates should not block productive use of the tool.

**Comment by MiiChielHD** [comment]
@Sam-VO I totally agree with the points you state.

@DariaMarkaryan @Destrolaric @devnaumov @E1izabeth @LenaVu @LonwoLonwo @m-ovs Sorry for tagging all of you, but this thread is already allmost 3 years going on with complaints about the update mechanism of DBeaver. We really like DBeaver, yet the update mechanism and the amount of updates is our point of frustration we hope you want to address. 

Could you give us any update on how you as developers of DBeaver are looking at this issue, is there any perspective on this being tackled in the near future?
